web/docs/database/operations/queries.md

```diff
@@ -0,0 +1,666 @@
+
+const MainPage = () => {
+  // TypeScript will automatically infer and type-check payload types.
+  const { data: allTasks, error: error1 } = useQuery(getAllTasks);
+  const { data: doneTasks, error: error2 } = useQuery(getFilteredTasks, {
+    isDone: true,
+  });
+
+  if (error1 !== null || error2 !== null) {
+    return <div>There was an error</div>;
+  }
+
+  return (
+    <div>
+      <h2>All Tasks</h2>
+      {allTasks && allTasks.length > 0
+        ? allTasks.map((task) => <Task key={task.id} {...task} />)
+        : "No tasks"}
+
+      <h2>Finished Tasks</h2>
+      {doneTasks && doneTasks.length > 0
+        ? doneTasks.map((task) => <Task key={task.id} {...task} />)
+        : "No finished tasks"}
+    </div>
+  );
+};
+
+const Task = ({ description, isDone }: Task) => {
+  return (
+    <div>
+      <p>
+        <strong>Description: </strong>
+        {description}
+      </p>
+      <p>
+        <strong>Is done: </strong>
+        {isDone ? "Yes" : "No"}
+      </p>
+    </div>
+  );
+};
+
+export default MainPage;
+```
+
+Notice how you don't need to annotate the Query's return value type. Wasp automatically infers the from the Query's backend implementation. This is **full-stack type safety**: the types on the client always match the types on the server.
+
+</TabItem>
+</Tabs>
+
+<small>
+
+For a detailed specification of the `useQuery` hook, check the [Options Reference](#options-reference).
+
+</small>
+
+### Error Handling
+
+For security reasons, all exceptions thrown in the Query's NodeJS implementation are sent to the client as responses with the HTTP status code `500`, with all other details removed.
+Hiding error details by default helps against accidentally leaking possibly sensitive information over the network.
+
+If you do want to pass additional error information to the client, you can construct and throw an appropriate `HttpError` in your implementation:
+
+<Tabs groupId="js-ts">
+<TabItem value="js" label="JavaScript">
+
+```js title=src/server/queries.js
+import HttpError from "@wasp/core/HttpError.js";
+
+export const getAllTasks = async (args, context) => {
+  throw new HttpError(
+    403, // status coce
+    "You can't do this!", // message
+    { foo: "bar" } // data
+  );
+};
+```
+
+</TabItem>
+<TabItem value="ts" label="TypeScript">
+
+```ts title=src/server/queries.ts
+import { GetAllTasks } from "@wasp/queries/types";
+import HttpError from "@wasp/core/HttpError.js";
+
+export const getAllTasks: GetAllTasks = async (args, context) => {
+  throw new HttpError(
+    403, // status coce
+    "You can't do this!", // message
+    { foo: "bar" } // data
+  );
+};
+```
+
+</TabItem>
+</Tabs>
+
+If the status code is `4xx`, the client will receive a response object with the corresponding `.message` and `.data` fields and rethrow the error (with these fields included).
+To prevent information leakage, the server won't forward these fields for any other HTTP status codes.
```

Comment:
```suggestion
If the status code is 4xx, the client will receive a response object with the corresponding .message and .data fields, and it will rethrow the error (including these fields). To prevent the leakage of information, the server won't forward these fields for any other HTTP status codes.
```